lib/widgets/lightbox.dart

```diff
@@ -113,6 +118,60 @@ class _CopyLinkButton extends StatelessWidget {
   }
 }
 
+class _DownloadImageButton extends StatelessWidget {
+  _DownloadImageButton({required this.url});
+
+  final downloader = ZulipBinding.instance.androidDownloadHost;
+  final Uri url;
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    return IconButton(
+      tooltip: zulipLocalizations.lightboxDownloadImageTooltip,
+      icon: const Icon(Icons.download),
+      onPressed: () async {
+        final scaffoldMessenger = ScaffoldMessenger.of(context);
+        String message = zulipLocalizations.lightboxDownloadImageFailed;
+        try {
+          await requestStoragePermission();
+          Uri downloadUrl = url;
+          final internalLink = parseInternalLink(url, store);
+          if (internalLink is UserUploadLink){
+            if (!context.mounted) return;
+            final tempUrl = await ZulipAction.getFileTemporaryUrl(context, internalLink);
+            downloadUrl = tempUrl ?? url;
+          }
+
+          final fileName = url.pathSegments.last.trim();
+          await downloader.downloadFile(downloadUrl.toString(), fileName);
+          message = zulipLocalizations.lightboxDownloadImageStarted;
+        } catch (e) {
+          if (e is TimeoutException || e is SocketException) {
+            message = zulipLocalizations.lightboxDownloadImageError;
+          } else {
+            message = zulipLocalizations.lightboxDownloadImageFailed;
+          }
```

Comment:
Could you explain why you chose

- "An error occurred while downloading the image." for `TimeoutException` or `SocketException`
- "Failed to download the image." for other errors

?